### ISSUE TITLE
Improve an error message from twcgi.py

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6881]
|[<img alt="fijal's avatar" src="https://avatars.githubusercontent.com/u/85944?s=50" width="50" height="50">](https://github.com/fijal)| @fijal reported|
|-|-|
|Trac ID|trac#6881|
|Type|enhancement|
|Created|2013-12-23 19:44:55Z|
|Branch|https://github.com/twisted/twisted/tree/improve-twcgi-error-message-6881|

changes:

Malformed CGI headers: message

to 

Malformed CGI headers: "message"

Makes it easier to spot problems related to newlines

Attachments:

* [twisted.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/0b4/0b458f92d4acd4b1ea79a58e7eac8f63f8839c0d/d28a367b58eba790fb131d7a60d9534697fcdd09.patch) (2450 bytes) - added by fijal on 2013-12-23 19:45:17Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__6881 6881
type__enhancement enhancement
reporter__fijal fijal
priority__normal normal
milestone__ 
branch__branches_improve_twcgi_error_message_6881 branches/improve-twcgi-error-message-6881
branch_author__jerith jerith
status__closed closed
resolution__fixed fixed
component__web web
keywords__ 
time__1387827895000000 1387827895000000
changetime__1388497478000000 1388497478000000
version__None None
owner__jerith jerith
cc__jknight
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

fijal would like this patch reviewed.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set owner to @fijal|
|-|-|

Thanks.

  1. Is `'"%s"' % (header,)` better than `'%r' % (header,)`?  It seems like the latter would make the output even more unambiguous.
  1. You may want to change this `log.msg` call to use the `format` style, as long as you're changing it: `log.msg(format='ignoring malformed CGI header: %(header)r', header=header)`
  1. `test_malformedHeaderCGI` adds a log observer to the global log publisher - but nothing ever removes it.  This is bad for a number of reasons.  The test should use `addCleanup` to make sure the observer is removed.
  1. This needs a news fragment too, but maybe it'll be a misc (therefore empty, therefore not a significant omission from the patch).

Thanks again!  These are all minor points so I'm happy to see this merged after they've been addressed (though you probably want to consult buildbot first to be safe).

**Comment by twisted-trac** [comment]
|[<img alt="jerith's avatar" src="https://avatars.githubusercontent.com/u/43731?s=50" width="50" height="50">](https://github.com/jerith)<a name="note_4"></a>|@jerith commented|
|-|-|

(In [41050]) Branching to 'improve-twcgi-error-message-6881'

**Comment by twisted-trac** [comment]
|[<img alt="jerith's avatar" src="https://avatars.githubusercontent.com/u/43731?s=50" width="50" height="50">](https://github.com/jerith)<a name="note_5"></a>|@jerith commented|
|-|-|

(In [41062]) Use format style logging, clean up observer in test, add news fragment.

refs [#6881](https://github.com/twisted/twisted/issues/6881)

**Comment by twisted-trac** [comment]
|[<img alt="jerith's avatar" src="https://avatars.githubusercontent.com/u/43731?s=50" width="50" height="50">](https://github.com/jerith)<a name="note_6"></a>|@jerith removed owner|
|-|-|

I've made the changes suggested in [comment:3 exarkun] but would like another (quick) review pass to make sure I didn't do anything silly.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun set owner to @jerith|
|-|-|

Thanks!

I forced a build of the branch to see if buildbot likes the code.  If that turns up green, please go ahead and merge.

Thanks again, fijal and jerith.

**Comment by twisted-trac** [comment]
|[<img alt="jerith's avatar" src="https://avatars.githubusercontent.com/u/43731?s=50" width="50" height="50">](https://github.com/jerith)<a name="note_8"></a>|@jerith set status to `closed`|
|-|-|

(In [41067]) Merge improve-twcgi-error-message-6881

Author: fijal, jerith
Reviewer: exarkun
Fixes: [#6881](https://github.com/twisted/twisted/issues/6881)

Log a better message for malformed CGI headers.
